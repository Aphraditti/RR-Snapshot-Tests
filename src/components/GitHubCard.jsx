
import { useState, useEffect } from "react";
import { Card } from "react-bootstrap"

export default function GitHubCard() {
    const [githubData, setGithubData] = useState({
        name: '',
        avatar_url: '',
        bio: ''
    });

    useEffect(() => {
        fetch('https://api.github.com/users/sfanetti')
            .then(res => res.json())
            .then(data => setGithubData(data))
    });

    return (
        <Card>
            <Card.Body>
                <Card.Img
                    variant="top"
                    src={ githubData.avatar_url }
                    alt=
                    "Github avatar"
                    />
                    <Card.Title>
                        <h2>{
                            githubData.name
                            }</h2>
                    </Card.Title>
                    <Card.Text>
                        {
                            githubData.bio
                        }
                    </Card.Text>
            </Card.Body>
        </Card>
    )

}