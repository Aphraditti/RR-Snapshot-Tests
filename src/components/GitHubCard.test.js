import { render, screen, waitFor } from "@testing-library/react";
import renderer from 'react-test-renderer';
import GitHubCard from "./GitHubCard";

const DEFAULT_MOCK_DATA = {
    name: 'ahiat',
    avatar_url: 'https://avatars.githubusercontent.com/u/117119563?v=4',
    bio: 'She is a smart and funny lady'

}

const DEFAULT_AVATAR_ALT = 'Github avatar'

describe('Test GitHubCard for features when', () => {
    beforeEach(() => {
        fetch.mockResponseOnce(JSON.stringify(DEFAULT_MOCK_DATA))
        render(<GitHubCard />);
    });
    afterEach(() => {
        fetch.resetMocks();
    })

    test('it containts a photo of the github user', async () => {
        const avatarImage = await waitFor(() => screen.getByAltText('Github avatar'));
        expect(avatarImage).toHaveAttribute('src',
            expect.stringContaining(DEFAULT_MOCK_DATA.avatar_url)
        );
    });

    test('it contains the name of the github user', async () => {
        const userName= await waitFor(() => screen.getByRole('heading', { level: 2 }));
        expect(userName).toHaveTextContent(DEFAULT_MOCK_DATA.name)
    });

    test('it contains a blurb about the github user', async () => {
        const blurb = await waitFor(() => screen.getByRole('paragraph'));
        expect(blurb).toHaveTextContent(DEFAULT_MOCK_DATA.bio);
    });
});

test('it renders a snapshot', () => {
    const tree = renderer.create(<GitHubCard/>).toJSON();
    expect(tree).toMatchSnapshot();
})
