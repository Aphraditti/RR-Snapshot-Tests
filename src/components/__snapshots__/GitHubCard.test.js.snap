// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <img
      alt="Github avatar"
      className="card-img-top"
      src=""
    />
    <div
      className="card-title h5"
    >
      <h2 />
    </div>
    <p
      className="card-text"
    />
  </div>
</div>
`;
